pinot-core/src/main/java/org/apache/pinot/core/query/reduce/BrokerReduceService.java

```diff
@@ -220,9 +221,9 @@ public BrokerResponseNative reduceOnDataTable(BrokerRequest brokerRequest,
 
     // Parse the option from request whether to preserve the type
     Map<String, String> queryOptions = brokerRequest.getQueryOptions();
-    String preserveTypeString =
-        (queryOptions == null) ? "false" : queryOptions.getOrDefault(QueryOptionKey.PRESERVE_TYPE, "false");
-    boolean preserveType = Boolean.valueOf(preserveTypeString);
+    boolean preserveType = QueryOptionsUtils.isPreserveType(queryOptions);
```

Comment:
This seems like cleanup items that are not really needed for using DataSchema in ResultTable. If so, just call it out in the PR description.

Reply:
Done